### PR TITLE
feat(VThemeProvider): add new component

### PR DESCRIPTION
**Pull request by johnleider**
## Description
Expose **ThemeProvider** as a standalone component, `v-theme-provider`.

## Motivation and Context
It's an easy addition and there have been times I know I've grabbed for it.

## How Has This Been Tested?
jest

## Markup:

<details>

```vue
<template>
  <div class="ma-12 pa-12">
    <v-theme-provider
      :dark="isDark"
      :light="!isDark"
    >
      <v-card :light="isComponentLight">
        <v-card-title>Title</v-card-title>

        <v-card-text>Lorem ipsum</v-card-text>

        <v-card-actions class="justify-center">
          <v-checkbox
            v-model="$vuetify.theme.dark"
            class="ma-1"
            label="Application Dark"
            hide-details
          ></v-checkbox>
          <v-checkbox
            v-model="isDark"
            class="ma-1"
            label="Theme Provider Dark"
            hide-details
          ></v-checkbox>
          <v-checkbox
            v-model="isComponentLight"
            class="ma-1"
            label="Component Light"
            hide-details
          ></v-checkbox>
        </v-card-actions>
      </v-card>
    </v-theme-provider>
  </div>
</template>

<script>
  export default {
    data: () => ({
      isComponentLight: false,
      isDark: false,
    }),

    computed: {
      isRootDark () {
        return this.$vuetify.theme.dark
      },
    },
  }
</script>

```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)
- [ ] I've added new examples to the kitchen (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by johnleider** [comment]
Requesting final comments @vuetifyjs/core-contributors @vuetifyjs/core-team

File: packages/vuetify/src/util/ThemeProvider.ts [hunk 1]
```diff
@@ -1,22 +1,16 @@
...
-    },
-  },
-
-  render (): VNode {
-    return this.$slots.default! && this.$slots.default!.find(node => !node.isComment && node.text !== ' ')!
+/* @vue/component */
+export default VThemeProvider.extend({
+  created () {
+    deprecate(
+      '<theme-provider></theme-provider>',
```
Comment (KaelWD) [inline packages/vuetify/src/util/ThemeProvider.ts]: This was never registered

File: packages/vuetify/src/components/VThemeProvider/VThemeProvider.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Mixins
+import Themeable from '../../mixins/themeable'
+
+// Types
+import { VNode } from 'vue'
+
+/* @vue/component */
+export default Themeable.extend({
+  name: 'theme-provider',
```
Comment (KaelWD) [inline packages/vuetify/src/components/VThemeProvider/VThemeProvider.ts]: v-

File: packages/vuetify/src/components/VThemeProvider/__tests__/VThemeProvider.spec.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  beforeEach(() => {
+    mountFunction = (options = {}) => {
+      return mount(VThemeProvider, {
+        ...options,
+      })
+    }
+  })
+
+  it('should work match snapshot', () => {
```
Comment (KaelWD) [inline packages/vuetify/src/components/VThemeProvider/__tests__/VThemeProvider.spec.ts]: Useless test

File: packages/vuetify/src/components/VThemeProvider/VThemeProvider.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        ? this.rootIsDark
+        : Themeable.options.computed.isDark.call(this)
+    },
+  },
+
+  render (): VNode {
+    /* istanbul ignore next */
+    return (
+      this.$slots.default! &&
+      this.$slots.default!.find(node => !node.isComment && node.text !== ' ')!
```
Comment (nekosaur) [inline packages/vuetify/src/components/VThemeProvider/VThemeProvider.ts]: why this find?

**Comment by nekosaur** [comment]
maybe add an example of its use? unless I'm missing it

**Comment by johnleider** [comment]
Unrelated to this PR, only the code format was changed.

**Comment by kamit-transient** [comment]
can this issue be taken as part of this PR [https://github.com/vuetifyjs/vuetify/issues/9873#issue-532482611](https://github.com/vuetifyjs/vuetify/issues/9873#issue-532482611) . This issue looks great candidate to be handled as part of theme provider it self.   Any Suggestions on this ?

**Pull request merged**